src/LodeRunner.Data/CosmosDBRepository.cs

```diff
@@ -202,10 +202,17 @@ public async Task<TEntity> CreateDocumentAsync<TEntity>(TEntity newDocument)
         /// <typeparam name="TEntity">The type of the entity.</typeparam>
         /// <param name="newDocument">The new document.</param>
         /// <param name="cancellationToken">The Cancellation Token.</param>
+        /// <param name="eTag">The entity tag associated with the resource.</param>
         /// <returns>An instance of the document.</returns>
-        public async Task<TEntity> UpsertDocumentAsync<TEntity>(TEntity newDocument, CancellationToken cancellationToken = default)
+        public async Task<TEntity> UpsertDocumentAsync<TEntity>(TEntity newDocument, CancellationToken cancellationToken = default, string eTag = "")
```

Comment:
might be better to switch etag here for ItemRequestOptions

Reply:
refactored